engine/apps/schedules/models/on_call_schedule.py

```diff
@@ -401,6 +401,66 @@ def _merge_events(self, events):
             events = merged
         return events
 
+    def _generate_ical_file_from_shifts(self, qs, extra_shifts=None, allow_empty_users=False):
+        raise NotImplementedError
+
+    def preview_shift(self, custom_shift, user_tz, starting_date, days, updated_shift_pk=None):
```

Comment:
FYI, I'm just moving this to the upper class (from `OnCallScheduleWeb` to `OnCallSchedule`), without any changes.